components/tenant-mgt/org.wso2.carbon.tenant.keystore.mgt/src/main/java/org/wso2/carbon/keystore/mgt/KeyStoreGenerator.java

```diff
@@ -160,26 +159,32 @@ private X509Certificate generateKeyPair(KeyStore keyStore) throws KeyStoreMgtExc
             String commonName = "CN=" + tenantDomain + ", OU=None, O=None L=None, C=None";
 
             //generate certificates
-            AlgorithmIdentifier sigAlgId = new DefaultSignatureAlgorithmIdentifierFinder().find("MD5WithRSAEncryption");
-            AlgorithmIdentifier digAlgId = new DefaultDigestAlgorithmIdentifierFinder().find(sigAlgId);
-            AsymmetricKeyParameter privateKeyAsymKeyParam = PrivateKeyFactory.createKey(keyPair.getPrivate().getEncoded());
-            SubjectPublicKeyInfo subPubKeyInfo = SubjectPublicKeyInfo.getInstance(keyPair.getPublic().getEncoded());
-            ContentSigner sigGen = new BcRSAContentSignerBuilder(sigAlgId, digAlgId).build(privateKeyAsymKeyParam);
+            X500Name distinguishedName = new X500Name(commonName);
+            X509CertInfo x509CertInfo = new X509CertInfo();
 
             Date notBefore = new Date(System.currentTimeMillis() - 1000L * 60 * 60 * 24 * 30);
             Date notAfter = new Date(System.currentTimeMillis() + (1000L * 60 * 60 * 24 * 365 * 10));
 
-            X509v3CertificateBuilder v3CertBuilder = new X509v3CertificateBuilder(new X500Name(commonName),
-                    BigInteger.valueOf(new SecureRandom().nextInt()),
-                    notBefore, notAfter, new X500Name(commonName), subPubKeyInfo);
+            CertificateValidity interval = new CertificateValidity(notBefore, notAfter);
+            BigInteger serialNumber = BigInteger.valueOf(new SecureRandom().nextInt());
 
-            X509CertificateHolder certificateHolder = v3CertBuilder.build(sigGen);
-            X509Certificate PKCertificate = new JcaX509CertificateConverter().setProvider("BC").getCertificate(certificateHolder);
+            x509CertInfo.set(X509CertInfo.VALIDITY, interval);
+            x509CertInfo.set(X509CertInfo.SERIAL_NUMBER, new CertificateSerialNumber(serialNumber));
+            x509CertInfo.set(X509CertInfo.SUBJECT, distinguishedName);
+            x509CertInfo.set(X509CertInfo.ISSUER, distinguishedName);
+            x509CertInfo.set(X509CertInfo.KEY, new CertificateX509Key(keyPair.getPublic()));
+            x509CertInfo.set(X509CertInfo.VERSION, new CertificateVersion(CertificateVersion.V3));
+
+            AlgorithmId signatureAlgoId = new AlgorithmId(AlgorithmId.md5WithRSAEncryption_oid);
+            x509CertInfo.set(X509CertInfo.ALGORITHM_ID, new CertificateAlgorithmId(signatureAlgoId));
+            PrivateKey privateKey = keyPair.getPrivate();
+            X509CertImpl x509Cert = new X509CertImpl(x509CertInfo);
+            x509Cert.sign(privateKey, "MD5withRSA", getPreferredJceProviderIdentifier());
```

Comment:
Did we use MD5withRSA in the previous implementation? This is a weak algorithm.

Reply:
Yes, this is used since it was used in the previous implementation